frameworks/helloworld/tests/test_sanity.py

```diff
@@ -196,6 +203,26 @@ def check_for_nonempty_properties():
     assert stdout == "true\n"
 
 
+@retrying.retry(
+    wait_fixed=10000,
+    stop_max_delay=120000,
+    retry_on_result=lambda res: res is False,
+    retry_on_exception=lambda ex: False if "failed: 409 Conflict" in ex.args[0] else True)
+def wait_for_refresh_cache_fails_409conflict():
+    """caching disabled, refresh_cache should fail with a
+    409 error (eventually, once scheduler is up)
+    """
+    sdk_cmd.svc_cli(config.PACKAGE_NAME, FOLDERED_SERVICE_NAME, 'state refresh_cache')
+    return False
```

Comment:
Is this not always going to retry? (I may be missing something with the lambdas)

Reply:
The intention of the code here is a little confusing:
* check the state refresh_cache command
* if it raises an exception (ANY exception), then we go on and check if its a 409. If it is, we exit the loop
* if its not a 409 exception then we continue

so how do we do that here? 
* We say `retry_on_result=lambda res: res is False,` so that non-exception raising runs simply retry (same as before)
* We say `retry_on_exception=lambda ex: False if "failed: 409 Conflict" in ex.args[0] else True` so that we always retry exceptions that are not the exected one.

However, there is a failure here and that is that once the expected 409 is raised, retrying.retry will raise it as an exception instead of suppressing it. So, we should either do:
```
try:
    wait_for_refresh_cache_fails_409conflict()
except retrying.RetryError:
    pass
```
However, this will also consider a timeout to be valid, so I think I need to take another look at the refactor. Most likely I need to add that `try:except` inside the retrying like the original